serializers/src/glTF/2.0/babylon.glTFExporter.ts

```diff
@@ -186,121 +216,39 @@ module BABYLON.GLTF2 {
          * @param strideSize - Represents the offset between consecutive attributes
          * @param byteOffset - The offset to start counting bytes from.
          * @param dataBuffer - The buffer to write the binary data to.
-         * @param useRightHandedSystem - Indicates whether the data should be modified for a right or left handed coordinate system. 
          * @returns - Byte length of the attribute data.
          */
-        private writeAttributeData(vertexBufferKind: string, meshAttributeArray: FloatArray, strideSize: number, vertexBufferOffset: number, byteOffset: number, dataBuffer: DataView, useRightHandedSystem: boolean): number {
+        private writeAttributeData(vertexBufferKind: string, meshAttributeArray: FloatArray, strideSize: number, vertexBufferOffset: number, byteOffset: number, dataBuffer: DataView): number {
             let byteOff = byteOffset;
 
-            const start = 0;
             const end = meshAttributeArray.length / strideSize;
 
             let byteLength = 0;
 
-            switch (vertexBufferKind) {
-                case VertexBuffer.PositionKind: {
-                    for (let k = start; k < end; ++k) {
-                        const index = k * strideSize!;
-                        dataBuffer!.setFloat32(byteOff, meshAttributeArray![index], true);
-                        byteOff += 4;
+            for (let k = 0; k < end; ++k) {
+                const index = k * strideSize;
+                dataBuffer.setFloat32(byteOff, meshAttributeArray[index], true);
+                byteOff += 4;
 
-                        dataBuffer!.setFloat32(byteOff, meshAttributeArray![index + 1], true);
-                        byteOff += 4;
+                dataBuffer.setFloat32(byteOff, meshAttributeArray[index + 1], true);
+                byteOff += 4;
+                if (strideSize > 2) {
+                    // Positions and Normals have 3 values.
+                    const scalez = (!this.convertToRightHandedSystem || vertexBufferKind === VertexBuffer.ColorKind) ? 1 : -1;

```

Comment:
What happens with the fourth component of tangents?

Reply:
I think I'll refactor this part.  The fourth component of the tangent should be negated, but it is overlooked in this implementation.

Reply:
I now use help functions to calculate the right handed vector for vector3 and vector4.